[NoQA] docs: clarify mobile setup steps in README (#70864)
Expensify/App: README.md

1. Create a GitHub Personal Access Token:
   - Go to [GitHub Settings > Developer Settings > Personal Access Tokens](https://github.com/settings/tokens)
   - Click "Generate new token (classic)"
   - Select the following scope:
     - `repo`

2. Copy the generated token

3. Add `GITHUB_TOKEN` to `.env` file with your generated token

### Running the mobile application 📱
* To install project dependencies run: `npm install`
* To start metro server run: `npm run start`
**Note:** For now this is a required step — metro needs to be called manually in a separate terminal.
* To run application on a **Development Simulator**: `npm run ios` or `npm run android`

After completing these steps, you should be able to start both mobile platform apps using the remote build.

### Troubleshooting
* If fetching cached builds for mobile fails, verify that both workflows in the GitHub repository have completed successfully. If they did not, Rock will be unable to fetch the cached builds. 

* If you’re still encountering errors, you can try running:
    - `git clean -fdx`
    - `git clean -fdx ./Mobile-Expensify`
    - You may also want to reinstall node_modules.

> * In the event of workflow failures, it is recommended to have the option to manually build the application. The following steps will cover the manual build process.  

Reviewer: this is a must, please make sure we include this guide

The following steps describe how to configure the project to fully utilize Rock.

### Generating GitHub Personal Access Token

To take advantage of remote builds, setup your GitHub Personal Access Token (PAT) in your `.env` file:

1. Create a GitHub Personal Access Token:
   - Go to [GitHub Settings > Developer Settings > Personal Access Tokens](https://github.com/settings/tokens)
   - Click "Generate new token (classic)"
   - Select the following scope:
     - `repo`

2. Copy the generated token

3. Add `GITHUB_TOKEN` to `.env` file with your generated token

### Running the mobile application 📱
* To install project dependencies run: `npm install`
* To start metro server run: `npm run start`
**Note:** For now this is a required step — metro needs to be called manually in a separate terminal.
* To run application on a **Development Simulator**: `npm run ios` or `npm run android`

After completing these steps, you should be able to start both mobile platform apps using the remote build.

### Troubleshooting
* Try re-installing dependencies:
    - `npm run i-standalone` for the standalone app
    - `npm install` for the hybrid app

> * Try running `npm run ios` or `npm run android` again

Reviewer 2: I'd mention different commands for hybrid and standalone (`ios-standalone` and `android-standalone`)

* If you’re still encountering errors, you can try running:
    - `git clean -fdx ios/` when running iOS
    - `git clean -fdx android/` when running Android
    - `git clean -fdx ./Mobile-Expensify` when running hybrid app

> * Then try running `npm run ios` or `npm run android` again

Reviewer 2: Same here

The following steps describe how to configure the project to fully utilize Rock.

### Generating GitHub Personal Access Token

To take advantage of remote builds, setup your GitHub Personal Access Token (PAT) in your `.env` file:

1. Create a GitHub Personal Access Token:
   - Go to [GitHub Settings > Developer Settings > Personal Access Tokens](https://github.com/settings/tokens)
   - Click "Generate new token (classic)"
   - Select the following scope:
     - `repo`

2. Copy the generated token

3. Add `GITHUB_TOKEN` to `.env` file with your generated token

### Running the mobile application 📱
* To install project dependencies run: `npm install`
* To start metro server run: `npm run start`
**Note:** For now this is a required step — metro needs to be called manually in a separate terminal.
* To run application on a **Development Simulator**:
    - For iOS `npm run ios-standalone` or `npm run ios` (hybrid app)
    - For Android `npm run android-standalone` or `npm run android` (hybrid app) 

After completing these steps, you should be able to start both mobile platform apps using the remote build.

> ### Troubleshooting

Reviewer: Let's also add this note at the beginning of the section:

If you haven't done any intentional edits outside of `src/` (like adding new dependencies) but your app is still running into a full build, remember that it's way easier to debug and address a remote cache miss rather than any compilation error.

* To run the **development web app**: `npm run web`
* Changes applied to Javascript will be applied automatically via WebPack as configured in `webpack.dev.ts`

## Running the mobile application using Rock 🪨

This project leverages [Rock](rockjs.dev) to manage builds. With Rock, the build is downloaded from a remote server first, and if a matching remote build cannot be found, it falls back to building locally. By storing the complete build on a remote server, Rock.js minimizes the need for local builds and enables streamlined setup through automated downloads.

The following steps describe how to configure the project to fully utilize Rock.

### Generating GitHub Personal Access Token

> To use rock remote builds for mobile applications, you need to have GitHub Personal Access Token in your `.env` file:

Reviewer: ```suggestion
To take advantage of remote builds, setup your GitHub Personal Access Token (PAT) in your `.env` file:
```
